### ISSUE TITLE
3GPP (bibxml5) mapping: "TS 24.293 0.3.0" does not exist

### ISSUE THREAD

**Issue by ronaldtse** [ietf-ribose/bibxml-service#175]
http://xml2rfc.tools.ietf.org/public/rfc/bibxml5/reference.3GPP.24.293.xml

```xml
<reference anchor="3GPP.24.293">
<front>
<title>ICS Protocol over USSD bearer</title>
<author>
<organization>3GPP</organization>
</author>
<date day="29" month="September" year="2008"/>
</front>
<seriesInfo name="3GPP TS" value="24.293 0.3.0"/>
<format type="HTML" target="http://www.3gpp.org/ftp/Specs/html-info/24293.htm"/>
</reference>
```

The reference is to `24.293 0.3.0`.

However, In 3GPP, TS 24.293 does not have a version of 0.3.0. The relevant link provided as:
https://portal.3gpp.org/desktopmodules/Specifications/SpecificationDetails.aspx?specificationId=1070

Click on "versions":
<img width="664" alt="Screenshot 2022-04-17 at 12 19 36 PM" src="https://user-images.githubusercontent.com/11865/163700229-74d249ae-c5b5-4579-8af1-aec12610e596.png">

https://www.3gpp.org/ftp/Specs/archive/24_series/24.293
<img width="610" alt="image" src="https://user-images.githubusercontent.com/11865/163700052-dac45118-3cca-4412-b009-bd2626e8e2f9.png">

FYI @rjsparks @TonyLHansen

_Originally posted by @ronaldtse in https://github.com/ietf-ribose/bibxml-service/issues/135#issuecomment-1100800008_

**Comment by ronaldtse** [comment]
This is the only missing reference in bibxml5.

**Comment by strogonoff** [comment]
I guess the question is would external tools break if xml2rfc paths like this are gone?

**Comment by ronaldtse** [comment]
You mean the one without a mapping?

My view. 
1. I think 3GPP references are rarely used in IETF. 
2. Incorrect references to nonexistent items should fail.

**Comment by strogonoff** [comment]
Okay. For the record, the  original design was based on making any existing path (wrong or right) working throughout the migration period (the period would be in effect for as long as xml2rfc-style compat API is in use; once external tools are updated they are presumed to switch to the new main API instead).

**Comment by ronaldtse** [comment]
Agree. My recommendation stands even if there was no BibXML service -- incorrect references should be removed from the data set at xml2rfc.

**Comment by rjsparks** [comment]
3gpp references are still frequently used (and were heavily used in the last two decades). I'm looking to see if there's a document in the archive that uses that reference.

**Comment by ronaldtse** [comment]
@rjsparks thanks for the clarification re: 3GPP being used, wrong assumption on my part.

**Comment by rjsparks** [comment]
The references I find in the rfc series have been handcrafted (didn't use the bibxml service). I'm not finding use of bibxml-3gpp/bibxml5 grepping the xml that was submitted, but there is a horde of xml that has never been submitted, and some that has (rightfully) already expanded it's references.

**Comment by rjsparks** [comment]
ah - above - the rpc may well have expanded the 3gpp references in the unprepped rfc xml for other reasons.

**Comment by ronaldtse** [comment]
That's comforting to know. In any case, is there any use of the particular reference "24.293"? If not, It can be safely removed.

**Comment by ronaldtse** [comment]
It might be better just to keep a note that "24.293" has been removed (as in an errata), and remove it since it is incorrect.

**Comment by rjsparks** [comment]
If we were to be as minimally invasive as possible, we could just add a comment to the returned result. However, if you don't remove it, it will show up in index searches. But I'm not sure how much harm that's going to do?

**Comment by ronaldtse** [comment]
We could handle this particular reference as a "deprecated" reference and return some data that says it was incorrect. (or is this a generic outcome that we should do with the other data sets with incorrect data, e.g. bibxml-ieee?)

When users sees incorrect references in search, it could potentially damage the confidence users have in the service that it always returns useful results...

**Comment by rjsparks** [comment]
I can find no use of this reference - we can remove it, and if we find that causes trouble, reconstitute it in -misc later

**Comment by ronaldtse** [comment]
Thank you @rjsparks ! Will get that done in the archive repo.

**Comment by ronaldtse** [comment]
Actually, I just realised a much better alternative -- we can just map "TS 24.293 0.3.0" to "TS 24.293 0.2.0", which is a lower version but exists. I have verified that the current bibxml5 dataset does not contain any other "24.293".

@rjsparks would that be the better solution?

**Comment by strogonoff** [comment]
> We could handle this particular reference as a "deprecated" reference and return some data that says it was incorrect.

We could indicate something via a special HTTP header… Not that most people will pay attention to that, of course, but it’ll at least be minimally invasive to actual XML contents, if that matters

**Comment by ronaldtse** [comment]
Yeah we could use an HTTP header to indicate deprecated data — in fact, maybe it is the case for all legacy paths!? Ie. We return the mapped data and indicate that this is the updated version.

**Comment by strogonoff** [comment]
We currently return an HTTP header that describes whether the item was manually mapped, automatically mapped or used fallback to xml2rfc data archive. If we provide a separate marker that says “this path is deprecated”, we can return that as well.

The important point is that we need to store this data somewhere. It’s basically adding a new metadata point about an xml2rfc path. (So far there’s only one such data point, and that is manual mapping for a path.)

So basically in that case we could generalize the xml2rfc path mappings storage to be more like “xml2rfc path metadata”, where metadata would consist of optional `docid` mapping and the new data point (e.g., optional `invalid` flag), and manage it in the same place (e.g., the Git repository provisioned per #181).

**Comment by strogonoff** [comment]
It’s important to think what we’re trying to achieve here. If we want people to *not use* deprecated paths, HTTP header is perhaps too subtle of indication, we’d be lucky if 1% of users ever notice. We could still keep these markers for future cleanup purposes though.

**Comment by ronaldtse** [comment]
> If we want people to stop using deprecated paths ASAP

I am not sure if the IETF community wants this to happen... but it should.

Maybe the only way to make that happen is for the data structure / contents to change.

And instead of returning a file, we can also do a redirect to the new API?

Anyway, for this particular reference, a mapping to the correct bibliographic item makes sense.

**Comment by rjsparks** [comment]
>@rjsparks would that be the better solution?

No - it really wouldn't. This is a concrete ask for a particular version and returning another would introduce more confusion than it would be helpful.

Just not finding the identifier would be better.

**Comment by ronaldtse** [comment]
@rjsparks got it, we will map the reference to a "remove" (will discuss with @strogonoff on the options).

**Comment by ronaldtse** [comment]
Implemented in #135.
